sklearn/utils/murmurhash.pyx

```diff
@@ -113,15 +121,15 @@ def murmurhash3_32(key, seed=0, positive=False):
             return murmurhash3_int_u32(<cnp.int32_t>key, seed)
         else:
             return murmurhash3_int_s32(<cnp.int32_t>key, seed)
-    elif isinstance(key, cnp.ndarray):
+    elif _is_arraylike(key):
```

Comment:
key must be a ndarray here since we check its dtype (list doesn't have dtype for instance)
```suggestion
    elif isinstance(key, np.ndarray):
```